src/Adapter/Category/Repository/CategoryRepository.php

```diff
@@ -140,6 +141,52 @@ public function assertCategoryExists(CategoryId $categoryId): void
         }
     }
 
+    /**
+     * Provides category names which are not unique per shop and language.
+     *
+     * e.g. if certain shop contains following categories in english language:
+     *      Clothes -> Men, Bags -> Men, Clothes -> Woman, Bags -> Women,
+     *      then method should return ["Men", "Women"]
+     *
+     * @param ShopId $shopId
+     * @param LanguageId $languageId
+     *
+     * @return string[]
+     */
+    public function getDuplicateNames(ShopId $shopId, LanguageId $languageId): array
+    {
+        $qb = $this->connection->createQueryBuilder()
+            ->select('cl.name, COUNT(cl.name) AS nameCount')
+            ->from($this->dbPrefix . 'category_lang', 'cl')
+            ->where('id_lang = :langId')
+            ->andWhere('id_shop = :shopId')
+            ->having('nameCount > 1')
+            ->setParameter('langId', $languageId->getValue())
+            ->setParameter('shopId', $shopId->getValue())
+            ->groupBy('cl.name')
+        ;
+
+        $results = $qb->execute()->fetchAllAssociative();
+
+        $names = [];
+        foreach ($results as $result) {
+            $names[] = $result['name'];
+        }
+
+        return $names;
+    }
+
+    /**
+     * @param CategoryId $categoryId
+     * @param LanguageId $languageId
+     *
+     * @return string[]
+     */
+    public function getBreadcrumbParts(CategoryId $categoryId, LanguageId $languageId): array
```

Comment:
Why adding a public method that only call a private. Why don't making this private public ?

Reply:
You're right I didn't notice but these three methods are kind of overkill A single public method should be enough

Reply:
I've already fixed it (got rid of one redundant private method). Not sure why you don't see it (unless there is something else Im missing?)